LOCAL DRAFT of WG Serving proposal

SergeyKanzhelev opened the pull request:
LOCAL DRAFT ONLY

YuikoTakada committed:
Add link to e2e_node test docs

johnbelamaric committed:
Add WG Device Management

Signed-off-by: John Belamaric <jbelamaric@google.com>

johnbelamaric committed:
Fix Kevin's github handle

xing-yang committed:
Update DP WG annual report 2023

sl1pm4t committed:
Add terraform-kops slack channel.

johnbelamaric committed:
Update charter

Signed-off-by: John Belamaric <jbelamaric@google.com>

smarterclayton commented on sigs.yaml:
```diff
@@ -3467,6 +3467,47 @@ workinggroups:
+- dir: wg-serving
+  name: Serving
+  mission_statement: >
+    Discuss and enhance the support of inference serving for accelerated workloads.
```
nit: stray period at end of line

smarterclayton commented on sigs.yaml:
```diff
@@ -3467,6 +3467,47 @@ workinggroups:
+  name: Serving
+  mission_statement: >
+    Discuss and enhance the support of inference serving for accelerated workloads.
+    in Kubernetes. Making Kubernetes the natural choice when choosing the platform
```
Suggestion for second sentence:

> Make Kubernetes the natural choice for hosting production inference reliably, and improve all serving workloads along the way.

smarterclayton commented on wg-serving/charter.md:
```diff
@@ -0,0 +1,83 @@
+### Out of scope
+
+- Batch inference, which should be covered by WG Batch.
+- Addition of new API kinds that serve a specialized type of workload. The focus
```
I don't know that this is out of scope.  Either way, it's vague.

ahg-g replied:
LWS is kinda that, so I don't think this should be out of scope

smarterclayton replied:
We should probably describe workflow for serving worlkoads as out of scope (i.e. we're not offering a MLOps platform, we're making Kubernetes better for the output of an MLOps pipeline)

SergeyKanzhelev replied:
added the workflows item. As for LWS - it is still generalized, no? I know it is written too vaguely and mostly copied from WG Batch charter. Changed to "Addition of new API kinds that serve a specific models. The focus should be on
  general APIs that frameworks can build on top of."

smarterclayton commented on wg-serving/charter.md:
```diff
@@ -0,0 +1,83 @@
+- Explore new projects that improve orchestration, scaling, and load balancing
+  of inference workloads and compose well with other workloads on Kubernetes
+
+### Out of scope
```
We need to spend more time calling out out of scope items

SergeyKanzhelev replied:
Some ideas, maybe too obvious to call out explicitly:

1. Maintain integrations with the specific frameworks
2. Security and privacy questions of the inference workloads beyond required extensibility points and monitoring capabilities
3. Questions of model testing and correctness beyond ability to deploy and execute test models.

SergeyKanzhelev replied:
workflows was a good call. Ease of integrating with plugins may be in scope?

smarterclayton commented on wg-serving/charter.md:
```diff
@@ -0,0 +1,83 @@
+Discuss and enhance the support for AI/ML inference workloads in Kubernetes.
+
+
+### In scope
```
Contrasting this to https://github.com/SergeyKanzhelev/community-1/blob/135df82737eba050d99268db7ed60b63baa12c9a/wg-batch/charter.md, i think we could tighten the in scope section up.  Maybe 1, 2, 4 could be one bullet, then we have another bullet with a suggestion of the areas that are in scope from the use cases doc?

ahg-g commented on sigs.yaml:
```diff
@@ -3467,6 +3467,47 @@ workinggroups:
+- dir: wg-serving
+  name: Serving
+  mission_statement: >
+    Discuss and enhance the support of inference serving for accelerated workloads.
```
Do we need to limit the focus to accelerated workloads only?

SergeyKanzhelev replied:
second sentence now covers other scenarios too. I think concentrating on accelerators would make sense. Are you thinking of CPU serving as a fallback option as a scenario? Or pure CPU serving that needs to be improved?

ahg-g commented on wg-serving/charter.md:
```diff
@@ -0,0 +1,83 @@
+
+### In scope
+
+- Provide concrete input to other SIGs and WG around needs of inference
```
```suggestion
- Provide concrete input to other SIGs and WGs around needs of inference
```

ahg-g commented on wg-serving/charter.md:
```diff
@@ -0,0 +1,83 @@
+Discuss and enhance the support for AI/ML inference workloads in Kubernetes.
+
+
+### In scope
```
mixing of batch/training and inference could be in scope, or do we want to leave that to wg-batch?

smarterclayton replied:
I think we should describe in scope as "being able to run serving workloads safely while giving up available slack capacity to batch frameworks"?

SergeyKanzhelev replied:
added

SergeyKanzhelev committed:
WG serving proposal

k8s-ci-robot committed:
Merge pull request #7726 from YuikoTakada/e2e_node-test

Add link to e2e_node test docs

k8s-ci-robot committed:
Merge pull request #7818 from sl1pm4t/slack-channel/terraform-kops

Add terraform-kops slack channel.

mrbobbytables committed:
Archive wg-iot-edge

wojtek-t committed:
Guidance for skipping Alpha for feature gated changes

soltysh committed:
Update sig-cli leads

k8s-ci-robot committed:
Merge pull request #7829 from soltysh/update_sig_cli

Update sig-cli leads

Priyankasaggu11929 committed:
sig-network: add kube-network-policies under network-policy subproject

k8s-ci-robot committed:
Merge pull request #7832 from Priyankasaggu11929/kube-network-policies

sig-network: add kube-network-policies under network-policy subproject

gjtempleton committed:
SIG-Autoscaling Leads Update

k8s-ci-robot committed:
Merge pull request #7827 from mrbobbytables/remove-iot-edge

Archive wg-iot-edge

justaugustus committed:
sig-release: Add inaugural subproject leads

Release Team:
- Grace Nguyen
- Kat Cosgrove

Release Engineering:
- Marko Mudrinić

Signed-off-by: Stephen Augustus <foo@auggie.dev>

k8s-ci-robot committed:
Merge pull request #7839 from justaugustus/sig-release-subproject-leads

sig-release: Add inaugural subproject leads

Priyankasaggu11929 committed:
sig-storage: remove container-object-storage-interface-csi-adapter for archival

rficcaglia committed:
Update sigs.yaml for wg-policy chair changes (#7643)

* Update sigs.yaml for wg-policy chair changes

wg-policy would like to add 2 new co-chairs

* generated from make generate per instructions

* Update sigs.yaml

added emeritus_leads per guidance

* Update sigs.yaml

apparently emeritus doesn't allow company field so I go incognito

* generated from make generate per instructions and emeritus change

johnbelamaric committed:
Meeting time has been determined

k8s-ci-robot committed:
Merge pull request #7828 from wojtek-t/feature_gates_criteria

Guidance for skipping Alpha for feature gated changes

johnbelamaric committed:
Fix roles and org management

k8s-ci-robot committed:
Merge pull request #7841 from Priyankasaggu11929/archive-container-object-storage-interface-csi-adapter

sig-storage: remove container-object-storage-interface-csi-adapter for archival

mrbobbytables committed:
sig-cloud-provider: remove baiducloud project

k8s-ci-robot committed:
Merge pull request #7843 from mrbobbytables/archive-repo

sig-cloud-provider: remove baiducloud project

k8s-ci-robot committed:
Merge pull request #7817 from xing-yang/data-protection-wg-report-2023

Update DP WG annual report 2023

johnbelamaric committed:
Rename Slack channel, update exit criteria

SergeyKanzhelev committed:
Added YouTube list creation in the list of action items

k8s-ci-robot committed:
Merge pull request #7849 from SergeyKanzhelev/patch-6

Added YouTube list creation in the list of action items

k8s-ci-robot committed:
Merge pull request #7837 from gjtempleton/SIG-Autoscaling-Leads-Update

SIG-Autoscaling Leads Update

mrbobbytables committed:
sig-auth: Add secrets-store-sync-controller subproject

k8s-ci-robot committed:
Merge pull request #7805 from johnbelamaric/wg-device-mgmt

Add WG Device Management

k8s-ci-robot committed:
Merge pull request #7853 from mrbobbytables/add-secrets-repo

sig-auth: Add secrets-store-sync-controller subproject
